popup.js

```diff
@@ -8,175 +8,117 @@ let translatorLanguage = 'fa';
-    'S': 'ئ',
-    'd': 'ی',
-    'D': 'ي',
-    'f': 'ب',
-    'F': 'إ',
-    'g': 'ل',
-    'G': 'أ',
-    'h': 'ا',
-    'H': 'آ',
-    'j': 'ت',
-    'J': 'ة',
-    'k': 'ن',
-    'K': '»',
-    'l': 'م',
-    'L': '«',
+    a: 'ش',
+    A: 'ؤ',
+    s: 'س',
+    S: 'ئ',
+    d: 'ی',
+    D: 'ي',
+    f: 'ب',
+    F: 'إ',
+    g: 'ل',
+    G: 'أ',
+    h: 'ا',
+    H: 'آ',
+    j: 'ت',
+    J: 'ة',
+    k: 'ن',
+    K: '»',
+    l: 'م',
+    L: '«',
     ';': 'ک',
     ':': ':',
-    '\'': 'گ',
+    "'": 'گ',
     '"': '؛',
-    'z': 'ظ',
-    'Z': 'ك',
-    'x': 'ط',
-    'X': 'ٓ',
-    'c': 'ز',
-    'C': 'ژ',
-    'v': 'ر',
-    'V': 'ٰ',
-    'b': 'ذ',
-    'B': '‌',
-    'n': 'د',
-    'N': 'ٔ',
-    'm': 'پ',
-    'M': 'ء',
+    z: 'ظ',
+    Z: 'ك',
+    x: 'ط',
+    X: 'ٓ',
+    c: 'ز',
+    C: 'ژ',
+    v: 'ر',
+    V: 'ٰ',
+    b: 'ذ',
+    B: '‌',
+    n: 'د',
+    N: 'ٔ',
+    m: 'پ',
+    M: 'ء',
     ',': 'و',
     '<': '<',
     '?': '؟',
 };
 
-const faToEnDictionary = {
-    'ض': 'q',
-    'ْ': 'Q',
-    'ص': 'w',
-    'ٌ': 'W',
-    'ث': 'e',
-    'ٍ': 'E',
-    'ق': 'r',
-    'ً': 'R',
-    'ف': 't',
-    'ُ': 'T',
-    'غ': 'y',
-    'ِ': 'Y',
-    'ع': 'u',
-    'َ': 'U',
-    'ه': 'i',
-    'ّ': 'I',
-    'خ': 'o',
-    ']': 'O',
-    'ح': 'p',
-    '[': 'P',
-    'ج': '[',
-    '}': '{',
-    'چ': ']',
-    '{': '}',
-    'ش': 'a',
-    'ؤ': 'A',
-    'س': 's',
-    'ئ': 'S',
-    'ی': 'd',
-    'ي': 'D',
-    'ب': 'f',
-    'إ': 'F',
-    'ل': 'g',
-    'أ': 'G',
-    'ا': 'h',
-    'آ': 'H',
-    'ت': 'j',
-    'ة': 'J',
-    'ن': 'k',
-    '»': 'K',
-    'م': 'l',
-    '«': 'L',
-    'ک': ';',
-    ':': ':',
-    'گ': '\'',
-    '؛': '"',
-    'ظ': 'z',
-    'ك': 'Z',
-    'ط': 'x',
-    'ٓ': 'X',
-    'ز': 'c',
-    'ژ': 'C',
-    'ر': 'v',
-    'ٰ': 'V',
-    'ذ': 'b',
-    '‌': 'B',
-    'د': 'n',
-    'ٔ': 'N',
-    'پ': 'm',
-    'ء': 'M',
-    'و': ',',
-    '<': '<',
-    '.': '.',
-    '>': '>',
-    '/': '/',
-    '؟': '?',
-};
+const faToEnDictionary = {};
+for (let key in enToFaDictionary) {
+    faToEnDictionary[enToFaDictionary[key]] = key;
+}
 
-const translator = (lang = 'fa') => str =>
-    str
-        .trim()
-        .split('')
-        .map(s => {
-            if (s === ' ') return ' ';
-            if (lang === 'fa') {
-                return enToFaDictionary.hasOwnProperty(s) ? enToFaDictionary[s] :
-                    faToEnDictionary.hasOwnProperty(s) ? faToEnDictionary[s] :
-                        s;
-            }
-            return faToEnDictionary.hasOwnProperty(s) ? faToEnDictionary[s] :
-                enToFaDictionary.hasOwnProperty(s) ? enToFaDictionary[s] :
-                    s;
-        }).join('');
+const translator =
+    (lang = 'fa') =>
+    str =>
+        str
+            .trim()
+            .split('')
+            .map(s => {
+                if (s === ' ') return ' ';
+                if (lang === 'fa') {
+                    return enToFaDictionary.hasOwnProperty(s)
```

Comment:
cause every correct in the dictionary has the truth value, what about changing it to something like this.

```js
if (lang === 'fa') return enToFaDictionary[s] || faToEnDictionary[s] || s

return faToEnDictionary[s] || enToFaDictionary[s] || s
```

Reply:
I agree with that. it's cleaner.👌🏻

Reply:
Sorry @amirHossein-Ebrahimi, I forget to say Hi :), and thanks for reviewing🙏🏻